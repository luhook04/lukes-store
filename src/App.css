.App {
  text-align: center;
}
img {
  width: 150px;
  height: 150px;
  border: 2px solid black;
}

nav a {
  font-weight: bold;
  font-size: 20px;
  color: white;
  text-decoration: none;
  padding: 20px;
}

.header {
  background-color: #2f4858;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 120px;
  margin-bottom: 75px;
  z-index: 1;
}

nav {
  column-gap: 70px;
  align-items: center;
  display: flex;
  justify-content: flex-end;
  width: 500px;
  padding-right: 150px;
}

h1 {
  color: #ff8066;
  font-size: 50px;
  flex-grow: 1;
  padding-left: 75px;
}

body {
  background-color: antiquewhite;
}

.welcome-message {
  color: #ff8066;
  font-size: 42px;
}

.home-container {
  text-align: center;
}

.store-btn {
  padding: 24px 48px;
  font-size: 20px;
  background-color: #2f4858;
  color: white;
  font-weight: bolder;
  border-radius: 5px;
}

.store-btn:hover {
  box-shadow: 0 12px 16px 0 rgba(0, 0, 0, 0.24),
    0 17px 50px 0 rgba(0, 0, 0, 0.19);
}

.product-grid {
  margin: 40px;
  text-align: center;
  display: grid;
  grid-template-columns: repeat(3, 1fr);

  border-radius: 5px;
}

.product-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  border: 1px solid black;
  background-color: #2f4858;
  color: white;
  font-size: 12px;
  padding: 20px;
}

.product-container h3 {
  margin-left: 5px;
  margin-right: 5px;
}

.product-container button {
  padding: 10px 18px;
}

.cart-grid {
  margin: 40px;
  display: grid;
  grid-template-columns: repeat(3, 1fr);

  border-radius: 5px;
}

.item-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  border: 1px solid black;
  background-color: #2f4858;
  color: white;
  font-size: 12px;
}

.item-container h2 {
  text-align: center;
  width: 90%;
  margin-left: auto;
  margin-right: auto;
}

.item-container img {
  margin-top: 45px;
}

.quantity {
  margin-bottom: -10px;
}

.item-container button {
  padding: 10px 16px;
  margin-left: 20px;
  margin-right: 20px;
}

.shopping-cart {
  text-align: center;
}

.shopping-cart button {
  margin-bottom: 75px;
  padding: 12px 18px;
  font-weight: bolder;
}

.shopping-cart h3 {
  font-size: 26px;
}

.empty-message {
  text-align: center;
}

.badge {
  font-size: 12px;
  border-radius: 50%;
  background-color: red;
  padding: 6px 7px;
  z-index: 1;
  position: absolute;
  right: 12px;
  top: 10px;
}

.cart-icon {
  position: relative;
  font-size: 35px;
  display: inline-block;
}

@media only screen and (max-width: 950px) {
  .product-grid {
    grid-template-columns: repeat(2, 1fr);
  }

  .cart-grid {
    grid-template-columns: repeat(2, 1fr);
  }

  .header {
    flex-direction: column;
    padding-bottom: 75px;
  }

  h1 {
    margin-top: 100px;
    text-align: center;
    padding-left: 0px;
  }

  nav {
    padding-right: 0px;
    justify-content: space-between;
  }
}

@media only screen and (max-width: 439px) {
  #root {
    width: fit-content;
  }

  nav {
    width: fit-content;
  }
}
