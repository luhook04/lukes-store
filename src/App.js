import "./App.css";
import React, { useState, useEffect } from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import Home from "./components/Home";
import Products from "./components/store/Products";
import Header from "./components/Header";
import ShoppingCart from "./components/cart/ShoppingCart";

const App = () => {
  const [ cart, setCart ] = useState([]);
  const [ store, setStore ] = useState([]);

  useEffect(() => {
    fetchStore();
  }, []);

  const fetchStore = async () => {
    const url = "https://fakestoreapi.com/products";
    const response = await fetch(url);
    const data = await response.json();
    setStore(data);
  };

  const addItem = (e) => {
    const item = {
      name  : e.target.parentNode.children[1].textContent,
      price : e.target.parentNode.children[2].textContent,
      image : e.target.parentNode.children[0].src
    };
    console.log(e.target.parentNode.children[0].src);
    setCart((prevCart) => [ ...prevCart, item ]);
  };

  return (
    <Router basename="/">
      <Header />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route
          path="/shop"
          element={<Products store={store} addItem={addItem} />}
        />
        <Route path="/cart" element={<ShoppingCart cart={cart} />} />
      </Routes>
    </Router>
  );
};

export default App;
