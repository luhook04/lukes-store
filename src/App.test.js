import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "./App";
// test that shopping cart render correct items on addItem(test in app.js)
