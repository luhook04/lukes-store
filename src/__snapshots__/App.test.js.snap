// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders header and home component 1`] = `
<div>
  <div>
    <h1>
      Store
    </h1>
    <nav>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/shop"
      >
        Products
      </a>
      <a
        href="/cart"
      >
        Cart
      </a>
    </nav>
  </div>
  <div>
    <h2>
      Welcome to the store
    </h2>
    <button>
      Shop Now!
    </button>
  </div>
</div>
`;

exports[`App component renders magnificent monkeys 1`] = `
<div>
  <div>
    <h1>
      Store
    </h1>
    <nav>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/shop"
      >
        Products
      </a>
      <a
        href="/cart"
      >
        Cart
      </a>
    </nav>
  </div>
  <div>
    <h2>
      Welcome to the store
    </h2>
    <button>
      Shop Now!
    </button>
  </div>
</div>
`;
